pkg/cidata/cidata.TEMPLATE.d/boot/15-pkg-cifs.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/sh
+set -eux
+
+# Install mount.cifs dependencies
+if [ "${LIMA_CIDATA_MOUNTS}" = 0 ] || command -v mount.cifs >/dev/null 2>&1; then
+	exit 0
+fi
+
+if command -v apt-get >/dev/null 2>&1; then
+	pkgs="cifs-utils"
+	if uname -a | grep -q -i "Ubuntu"; then
+		# install nls_utf8.ko for mounting cifs with iocharset=utf8
+		# This module is installed by default on Debian, but not by default on Ubuntu.
+		pkgs="${pkgs} linux-modules-extra-$(uname -r)"
+	fi
+	DEBIAN_FRONTEND=noninteractive
+	export DEBIAN_FRONTEND
+	apt-get update
+	# shellcheck disable=SC2086
+	apt-get install -y -q $pkgs
+elif command -v dnf >/dev/null 2>&1; then
+	dnf install -y cifs-utils
+elif command -v pacman >/dev/null 2>&1; then
+	pacman -Syu --noconfirm cifs-utils
+elif command -v zypper >/dev/null 2>&1; then
+	zypper install -y cifs-utils
+	# openSUSE 15.3 doesn't seem to provide nls_utf8.ko, though /proc/config.gz contains CONFIG_NLS_UTF8=m
+	# https://bugzilla.opensuse.org/show_bug.cgi?id=1190797
```

Comment:
@jandubois Any chance to get your colleagues to look into this? 🙏

Reply:
> @jandubois Any chance to get your colleagues to look into this? 🙏

Trying to find somebody, but they are in a different timezone. But I've send some requests...

Reply:
Thank you!

Reply:
The issue is not openSUSE, but that we are using the JeOS image:

>Because of size constraints, JeOS images use the `kernel-default-base` as the default kernel. Since the kernel does not contain drivers for bare metal, this allows for a smaller footprint.
>If a specific usage scenario requires any of the omitted modules, install the `kernel- default` package to get a full tree of the kernel modules. This is the same kernel used by a regular SUSE Linux Enterprise Server installation.

The problem is that installing `kernel-default` requires a reboot. 😞 

Will see if there are any regular openSUSE images with cloud-init support.

Reply:
Would this be a block for this PR, since other distributions can be satisfied? We could first drop the openSUSE support and later add it back, wonder how many ppl are relying on the openSUSE image now.

Reply:
I'm getting the same error with the default Ubuntu image:

```console
root@lima-default:~# modprobe nls_utf8
modprobe: FATAL: Module nls_utf8 not found in directory /lib/modules/5.11.0-31-generic
```

What am I missing?

Reply:
I could get it to work with this provisioning script:

```yaml
provision:
- mode: system
  script: |
    #!/bin/sh
    set -o errexit -o nounset -o xtrace
    if ! modprobe nls_utf8; then
      zypper in -y kernel-default
      reboot
    fi
```

Of course this is a bit wonky, as it reboots too late, so you may already connect to the instance. But I'm sure this could be moved into the boot script as well and would be mostly invisible. Maybe this is good enough?

Reply:
Thanks, rebooting is okay, but I guess we should make samba non-default then.

Reply:
> Thanks, rebooting is okay, but I guess we should make samba non-default then.

idk, it is a one-time thing during first boot only. So if samba is otherwise much better, I think we could still make it the default.

Was there a way to make the mounting of the file systems faster? It seemed like the first mount took a very long time; the second one was much faster.

Reply:
> It seemed like the first mount took a very long time; the second one was much faster.

That’s another reason that we should make samba optional 😞